modules/exploits/unix/webapp/spip_rce_form.rb

```diff
@@ -83,38 +84,20 @@ def check
     res = send_request_cgi({ 'uri' => uri.to_s })
 
     return Exploit::CheckCode::Unknown('Target is unreachable.') unless res
-    return Exploit::CheckCode::Unknown("Target responded with unexpected HTTP response code: #{res.code}") unless res.code == 200
 
-    version_string = res.get_html_document.at('head/meta[@name="generator"]/@content')&.text
-    return Exploit::CheckCode::Unknown('Unable to find the version string on the page: spip.php') unless version_string =~ /SPIP (.*)/
+    rversion = spip_version
```

Comment:
```
msf6 exploit(unix/webapp/spip_rce_form) > check http://127.0.0.1:8000

[-] Exploit failed: ArgumentError Malformed version number string SPIP 4.2.12
[-] 127.0.0.1:8000 - Check failed: The state could not be determined.
```
Seems like this is pulling out `SPIP 4.2.12` for me in this module and Rex:Version does not like that
I replicated on the other module too, not sure if this is a version specific thing or not

Reply:
You're absolutely right, I didn't test the PR well enough. I just pushed some changes that should fix this.

Reply:
```
msf6 exploit(unix/webapp/spip_rce_form) > check http://127.0.0.1:8000

[-] Exploit failed: ArgumentError Malformed version number string 4.2.12 @ www.spip.net +
[-] 127.0.0.1:8000 - Check failed: The state could not be determined.
```
getting a slightly different error now

Reply:
I'm confused as why the regex (`/SPIP (.*) /`) doesn't match. I don't have a testing environment handy (holidays, yay!), but something like `/SPIP ([0-9.]+) /` might fix the issue?

Reply:
I think we want `/SPIP (.*)/` seems like there was a rogue space messing things up
```
[7] pry(#<Msf::Modules::Exploit__Unix__Webapp__Spip_rce_form::MetasploitModule>)> version_string
=> "SPIP 4.2.12"
[8] pry(#<Msf::Modules::Exploit__Unix__Webapp__Spip_rce_form::MetasploitModule>)> version_string =~ /SPIP (.*) /
=> nil
[9] pry(#<Msf::Modules::Exploit__Unix__Webapp__Spip_rce_form::MetasploitModule>)> version_string =~ /SPIP (.*)/
=> 0
[10] pry(#<Msf::Modules::Exploit__Unix__Webapp__Spip_rce_form::MetasploitModule>)> ::Regexp.last_match(1)
=> "4.2.12"
```

Reply:
Nice catch! Care to fix it so this can land, or do you prefer to wait until I get back home?

Reply:
I can do it no problem

Reply:
I'll owe your a beverage of your choice shall we ever meet in the meatspace then <3